src/main/java/duck/storage/LocalStorage.java

```diff
@@ -0,0 +1,75 @@
+package duck.storage;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
+import duck.exception.DuckException;
+import duck.task.TaskList;
+
+/**
+ * LocalStorage makes use of Serializable objects to write to a local file.
+ */
+public class LocalStorage implements Storage {
+    private String filePath;
+
+    /**
+     * Initializes file location.
+     *
+     * @param filePath Location that storage file will be.
+     */
+    public LocalStorage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    private void ensureFileExists() throws IOException {
+        File file = new File(this.filePath);
+        if (!file.exists()) {
+            File directories = new File(file.getParentFile().getAbsolutePath());
+            directories.mkdirs();
+            file.createNewFile();
+        }
+
+    }
+
+
+    /**
+     * Serializes the TaskList object and stores in a file.
+     *
+     * @param taskList List of tasks to be saved.
+     * @throws DuckException If file is unable to be saved due to corruption.
+     */
+
+    public void save(TaskList taskList) throws DuckException {

```

Comment:
Maybe leave the javadoc right above the method declaration would make it look nicer.

Reply:
Woops, thanks for catching this. I'll fix it